### ISSUE TITLE
burn-tch: backward() on a zero-sized tensor panics in the in-place binary-op fast path

### ISSUE THREAD

**Issue by Yashiru** [tracel-ai/burn#5287]
### Describe the bug

Any `backward()` whose autodiff graph contains a zero-sized (empty) root panics on the `burn-tch`
(LibTorch) backend. During the backward pass an elementwise binary op runs between an empty `[0]`
tensor and a `[1]` gradient, and LibTorch rejects it:

```
Torch("output with shape [1] doesn't match the broadcast shape [0]")
```

The same computation passes on `ndarray`.

### Root cause

`TchTensor::binary_ops_tensor` (`crates/burn-tch/src/tensor.rs`) derives the output shape
dimension-wise with `usize::max`:

```rust
out_shape[i] = usize::max(lhs_shape[i], rhs_shape[i]); // tensor.rs:280
```

For a `[0]` vs `[1]` broadcast the correct result is `[0]`, but `max(0, 1)` gives `[1]`. With
`num_elements_out == 1`, the `rhs` (the `[1]` gradient) is treated as already matching the output
element count, so the in-place fast path is taken:

```rust
|lhs, rhs| rhs.f_mul_(lhs).unwrap(), // crates/burn-tch/src/ops/base.rs:472
```

LibTorch then computes the real broadcast (`[0]`) and refuses to write it into the `[1]` buffer.

### To Reproduce

Minimal, in plain torch:

```python
torch.ones(1).mul_(torch.ones(0))  # same error
```

In burn (no `mask_select` involved), on `Autodiff<LibTorch>`:

```rust
tensor.select(0, empty_indices).sum().backward()
```

It surfaces via `test_mask_select_grad_empty_mask` (added in #5238, promoted in #5283): a `mask_select`
on an all-`false` mask produces the empty `[0]` tensor that feeds this path. `mask_select` is not the
cause; it only creates the empty tensor.

### Expected behavior

`backward()` on a zero-sized root should produce an all-zero gradient (as it does on `ndarray`),
not panic.

### Proposed fix

In `TchTensor::binary_ops_tensor`, either:

- derive a `0` output dim as `0` rather than `usize::max` when either operand's dim is `0`, or
- skip the in-place fast paths when either operand has zero elements.

### Additional context

Found while promoting `mask_select` to a backend op in #5283; filed separately as suggested in that
PR's review.

I'd be happy to take this on and open a fix PR along the lines suggested above.
